### ISSUE TITLE
Catch exceptions in main()?

### ISSUE THREAD

**Issue by elfring** [rlguy/GridFluidSim3D#9]
I expect that exception handling is usually supported by a C++ program. I wonder why your function "[main](https://github.com/rlguy/GridFluidSim3D/blob/dfeb22532cd2d9ae0e4c0f580558f3868621f9b2/src/main.cpp#L3)" does not contain [corresponding try and catch instructions](http://stackoverflow.com/questions/368184/does-it-make-sense-to-catch-exceptions-in-the-main) so far.

How do you think about recommendations by Matthew Wilson in [an article](http://accu.org/index.php/journals/1706)?

Would you like to adjust the implementation if you consider [effects for uncaught/unhandled exceptions](http://www.informit.com/guides/content.aspx?g=cplusplus&seqNum=108) like they are described by Danny Kalev?

**Comment by rlguy** [comment]
Hello elfring,

Thanks for providing those links, they were interesting and I learned a few things!

I will be leaving the [main](https://github.com/rlguy/GridFluidSim3D/blob/dfeb22532cd2d9ae0e4c0f580558f3868621f9b2/src/main.cpp#L3) function as it is. I meant for the main method to contain a simple example for how to use the FluidSimulation class and I would like to keep that method as uncluttered as possible. The user is free to add in exception/error handling when using the FluidSimulation class themselves.

I did modify the public methods in the FluidSimulation class to throw exceptions on invalid input instead of triggering assertions.

Best Regards,
Ryan
